File: Technical_test_CAP_frontend/frontend/src/app/services/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Order } from '../model/Order';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  private url = "http://localhost:8080/api/orders";

  constructor(private http: HttpClient) {}

  createOrder(order: Order): Observable<Order> {
    return this.http.post<Order>(this.url, order);
  }

  getAllOrdersByUser(idUser: number): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.url}/user/${idUser}`);
  }

  getOrderById(id: number): Observable<Order> {
    return this.http.get<Order>(`${this.url}/${id}`);
  }

  updateOrder(id: number, order: Order): Observable<Order> {
    return this.http.put<Order>(`${this.url}/${id}`, order);
  }

  deleteOrder(id: number): Observable<void> {
    return this.http.delete<void>(`${this.url}/${id}`);
  }

  findAllByNumberOrder(numberOrder: string): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.url}/number/${numberOrder}`);
  }
}

File: Technical_test_CAP/src/main/java/com/fractal/technical_test_cap/Service/OrderService.java
package com.fractal.technical_test_cap.Service;

import com.fractal.technical_test_cap.DTO.OrderDTO;

import java.util.List;

public interface OrderService {
    OrderDTO saveOrder(OrderDTO orderDTO);

    List<OrderDTO> findAllOrdersByUser(Long idUser);

    OrderDTO updateOrder (OrderDTO orderDTO);

    void deleteOrder(Long id);

    OrderDTO getOrderById(Long id);

    List<OrderDTO> findAllByNumberOrder(String numberOrder);
}
